src/injector.js

```diff
@@ -0,0 +1,69 @@
+/** Copyright (c) 2018 Uber Technologies, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import * as React from 'react';
+
+import FusionApp, {createPlugin} from 'fusion-core';
+
+type Dependencies = {[string]: any};
+type Services = {[string]: any};
+
+const InjectorContext = React.createContext(
+  (tokens: Dependencies): Services => ({})
+);
+
+function getServices(app: FusionApp, deps: Dependencies): Services {
+  const serviceMap = {};
+
+  Object.entries(deps).forEach(([name, token]) => {
+    const ref = (token && token.ref) || token;
+
+    serviceMap[name] = app.services.get(ref);
+  });
+
+  return serviceMap;
+}
+
+export function registerInjector(app: FusionApp) {
+  function inject(deps: Dependencies): Services {
+    return getServices(app, deps);
+  }
+
+  const injectorPlugin = createPlugin({
+    middleware: () => (ctx, next) => {
+      ctx.element = (
```

Comment:
This should check for `ctx.element` before wrapping it. `ctx.element` is only present on server side renders